libecl_well/src/well_conn.c

```diff
@@ -211,9 +220,14 @@ well_conn_type * well_conn_alloc_from_kw( const ecl_kw_type * icon_kw ,
       connection_factor = ecl_kw_iget_as_double(scon_kw , scon_offset + SCON_CF_INDEX);
     }
 
-    {
-      int segment_id = ecl_kw_iget_int( icon_kw , icon_offset + ICON_SEGMENT_INDEX ) - ECLIPSE_WELL_SEGMENT_OFFSET + WELL_SEGMENT_OFFSET;
-      conn = well_conn_alloc__(i,j,k,connection_factor,dir,open,segment_id,matrix_connection);
+    int segment_id = ecl_kw_iget_int( icon_kw , icon_offset + ICON_SEGMENT_INDEX ) - ECLIPSE_WELL_SEGMENT_OFFSET + WELL_SEGMENT_OFFSET;
+    conn = well_conn_alloc__(i,j,k,connection_factor,dir,open,segment_id,matrix_connection);
+
+    if (xcon_kw) {
+      const int xcon_offset = header->nxconz * (header->ncwmax * well_nr + conn_nr);
+      conn->water_rate = ecl_kw_iget_as_double(xcon_kw, xcon_offset + XCON_WRAT_INDEX);
```

Comment:
The `XCON` keyword *is* double and the get function can be simply: `ecl_kw_iget_double( xcon_kw , ... )`.